src/main/kotlin/org/jitsi/nlj/dtls/DtlsStack.kt

```diff
@@ -146,6 +150,10 @@ class DtlsStack(
      */
     fun start() {
         roleSet.await()
+        synchronized(lock) {
+            running = true
+            // TODO: could updating dtlsTransport in this lock solve the issue below?
```

Comment:
I'm pretty sure you don't want to, `start()` is blocking.

Reply:
Not sure I follow you, we call `dtlsTransport = role?.start()` in `start` regardless of whether or not it's in the lock, I was just wondering if we put it inside the lock we could get rid of the call to `processIncomingProtocolData` here.  But these changes would also change the 'race' behavior: before we wouldn't drop the packet, but now if the packet comes before `running` is set to true, we'll drop it.

Reply:
My point is that if we called `dtlsTransport = role?.start()` inside a lock, we'd be holding the lock for an extended period  of time, and thus blocking whatever thread calls `processIncomingProtocolData`, which seems like a bad idea.

Reply:
The overlap there isn't as bad as it may seem: `role?.start()` is what establishes the connection, so we can't receive anything until that is established anyway.  

The previous issue was that the far end could see the connection as established before the `dtlsTransport = role?.start()` assignment completely finished and therefore a packet could come into the public `processIncomingProtocolData` in the window between the far side seeing the connection as being established and this side seeing the same thing and finishing the assignment.  

If we move `role?.start()` then the public `processIncomingProtocolData` would block until this finished which, in this case, would be by design: then there's no need to call (the private) `processIncomingProtocolData` after this line finishes, because the other code would be waiting on the lock and when it runs it would process as normal.  So it gets rid of the race condition that required this extra call to `processIncomingProtocolData`.

Reply:
Right; my worry is what blocking the thread which calls `processIncomingProtocolData` will do - does it back up other data behind this?

Reply:
It's the IceTransport reader IO thread that calls this, but there would only be contention here the very first time after DTLS completes, so traffic won't there wouldn't have been ongoing traffic there already.